packages/rust/server/src/models/group.rs

```diff
@@ -0,0 +1,241 @@
+}
+
+pub async fn get_user_groups(db: &DbPool, user_id_in: i32) -> Result<Vec<UserGroup>, sqlx::Error> {
+    let mut user_groups: Vec<UserGroup> = vec![];
+    let row = sqlx::query("SELECT * FROM user_group WHERE user_id = $1")
+        .bind(user_id_in)
+        .fetch_all(db)
+        .await?;
+    for i in row {
+        let role_id: i32 = i.get("role_id");
+        let user_role = match Role::try_from(role_id as u8) {
+            Ok(user_role) => user_role,
+            Err(_) => return Err(sqlx::Error::RowNotFound),
+        };
+        let new_user_group = UserGroup {
+            id: i.get("id"),
+            user_id: user_id_in,
+            group_id: i.get("group_id"),
+            role_id: user_role,
+            created: i.get("created"),
+        };
+        user_groups.push(new_user_group);
+    }
+    Ok(user_groups)
+}
+
+impl Group {
+    pub async fn role(&self, db: &DbPool, user_id: i32) -> Result<Role, sqlx::Error> {
+        let row = sqlx::query("SELECT * FROM user_group WHERE group_id = $1 AND user_id = $2;")
+            .bind(self.id)
+            .bind(user_id)
+            .fetch_one(db)
+            .await?;
+        let role_id: i32 = row.get("role_id");
+        let role_enum = match Role::try_from(role_id as u8) {
+            Ok(role_enum) => role_enum,
+            Err(_) => return Err(sqlx::Error::RowNotFound),
+        };
+        Ok(role_enum)
+    }
+
+    pub async fn members(&self, db: &DbPool) -> Result<Vec<User>, sqlx::Error> {
+        let row = sqlx::query("SELECT * FROM user_group WHERE group_id = $1 ")
+            .bind(self.id)
+            .fetch_all(db)
+            .await?;
+        let mut users_in_group: Vec<User> = vec![];
+        for i in row {
+            let user_to_be_added = User::from_id(db, i.get("id")).await?;
+            users_in_group.push(user_to_be_added);
+        }
+        Ok(users_in_group)
+    }
+
+    pub fn from_row(row: &PgRow) -> Group {
+        Group {
+            id: row.get("id"),
+            uuid: row.get("uuid"),
+            group_name: row.get("group_name"),
+            created: row.get("created"),
+            members: Vec::new(),
+        }
+    }
+
+    pub async fn from_uuid(db: &DbPool, uuid: Uuid) -> Result<Group, sqlx::Error> {
+        let row = sqlx::query("SELECT * FROM \"group\" WHERE uuid = $1")
+            .bind(uuid)
+            .fetch_one(db)
+            .await?;
+        let group = Self::from_row(&row);
+        Ok(group)
+    }
+
+    pub async fn from_id(db: &DbPool, id: i32) -> Result<Group, sqlx::Error> {
+        let row = sqlx::query("SELECT * FROM \"group\" WHERE id = $1")
+            .bind(id)
+            .fetch_one(db)
+            .await?;
+        let group = Self::from_row(&row);
+        Ok(group)
+    }
+
+    pub async fn create(&mut self, db: &DbPool, user_id: i32) -> Result<(), sqlx::Error> {
+        let row =
+            sqlx::query("INSERT INTO \"group\" (uuid, group_name) VALUES ($1, $2) RETURNING *;")
```

Comment:
lets do this in one query in one transaction

Reply:
done